Index: kdeartwork-14.11.95/CMakeLists.txt
===================================================================
--- kdeartwork-14.11.95.orig/CMakeLists.txt
+++ kdeartwork-14.11.95/CMakeLists.txt
@@ -18,9 +18,9 @@ macro_optional_add_subdirectory(ColorSch
 macro_optional_add_subdirectory(IconThemes)
 macro_optional_add_subdirectory(emoticons)
 macro_optional_add_subdirectory(kscreensaver)
-if (Q_WS_X11)
-   macro_optional_add_subdirectory(kwin-styles)
-endif (Q_WS_X11)
+#if (Q_WS_X11)
+#   macro_optional_add_subdirectory(kwin-styles)
+#endif (Q_WS_X11)
 macro_optional_add_subdirectory(styles)
 macro_optional_add_subdirectory(wallpapers)
 macro_optional_add_subdirectory(HighResolutionWallpapers)
